From dc0d71e2b8f9dc78709abd8be0dc75aa4f0aebb2 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Wed, 8 Jul 2015 15:08:32 +0200
Subject: [PATCH 56/60] common: set Iridium version and change User-Agent
 accordingly

Reduced version number:
We do not use @BUILD@ or @PATCH@, in particular not in the User-Agent
string.

As for the user agent, continue providing Chrome/* for possible
compatibility checks by the browser and/or websites, since we
really are still Chromium.

Note to self:
Update "Chromium/*" in UA string when updating chrome/version.
---
 chrome/VERSION                                            | 8 ++++----
 components/version_info/version_info.cc                   | 2 +-
 components/version_info/version_info_values.h.version     | 2 +-
 content/common/user_agent.cc                              | 2 ++
 extensions/shell/common/version.h.in                      | 2 +-
 tools/metrics/histograms/extract_histograms.py            | 2 +-
 .../histograms/generate_expired_histograms_array.py       | 4 ++--
 7 files changed, 12 insertions(+), 10 deletions(-)

diff --git a/chrome/VERSION b/chrome/VERSION
index 3248f1ae9104..99639d9efce1 100644
--- a/chrome/VERSION
+++ b/chrome/VERSION
@@ -1,4 +1,4 @@
-MAJOR=71
-MINOR=0
-BUILD=3578
-PATCH=30
+MAJOR=2018
+MINOR=11
+BUILD=71
+PATCH=0
diff --git a/components/version_info/version_info.cc b/components/version_info/version_info.cc
index f378da9a6857..d30c8b6149a6 100644
--- a/components/version_info/version_info.cc
+++ b/components/version_info/version_info.cc
@@ -13,7 +13,7 @@
 namespace version_info {
 
 std::string GetProductNameAndVersionForUserAgent() {
-  return "Chrome/" + GetVersionNumber();
+  return "Iridium/" + GetVersionNumber();
 }
 
 std::string GetProductName() {
diff --git a/components/version_info/version_info_values.h.version b/components/version_info/version_info_values.h.version
index 6232e359ef20..3cf4fee096c0 100644
--- a/components/version_info/version_info_values.h.version
+++ b/components/version_info/version_info_values.h.version
@@ -6,7 +6,7 @@
 #define COMPONENTS_VERSION_INFO_VERSION_INFO_VALUES_H_
 
 #define PRODUCT_NAME "@PRODUCT_FULLNAME@"
-#define PRODUCT_VERSION "@MAJOR@.@MINOR@.@BUILD@.@PATCH@"
+#define PRODUCT_VERSION "@MAJOR@.@MINOR@"
 #define LAST_CHANGE "@LASTCHANGE@"
 #define IS_OFFICIAL_BUILD @OFFICIAL_BUILD@
 
diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
index 0ceb7311a0f6..21c0121ef6e5 100644
--- a/content/common/user_agent.cc
+++ b/content/common/user_agent.cc
@@ -191,6 +191,8 @@ std::string BuildUserAgentFromOSAndProduct(const std::string& os_info,
       product.c_str(),
       WEBKIT_VERSION_MAJOR,
       WEBKIT_VERSION_MINOR);
+  if (strncmp(product.c_str(), "Chrome/", 7) != 0)
+	base::StringAppendF(&user_agent, " Chrome/71.0.0.0");
   return user_agent;
 }
 
diff --git a/extensions/shell/common/version.h.in b/extensions/shell/common/version.h.in
index c4627fe512a1..bf7d27484e99 100644
--- a/extensions/shell/common/version.h.in
+++ b/extensions/shell/common/version.h.in
@@ -7,7 +7,7 @@
 #ifndef EXTENSIONS_SHELL_COMMON_VERSION_H_
 #define EXTENSIONS_SHELL_COMMON_VERSION_H_
 
-#define PRODUCT_VERSION "@MAJOR@.@MINOR@.@BUILD@.@PATCH@"
+#define PRODUCT_VERSION "@MAJOR@.@MINOR@"
 #define LAST_CHANGE "@LASTCHANGE@"
 
 #endif  // EXTENSIONS_SHELL_COMMON_VERSION_H_
diff --git a/tools/metrics/histograms/extract_histograms.py b/tools/metrics/histograms/extract_histograms.py
index c18e4ab126a9..3eae82d2ff3a 100644
--- a/tools/metrics/histograms/extract_histograms.py
+++ b/tools/metrics/histograms/extract_histograms.py
@@ -69,7 +69,7 @@ DEFAULT_BASE_HISTOGRAM_OBSOLETE_REASON = (
     'Base histogram. Use suffixes of this histogram instead.')
 
 EXPIRY_DATE_PATTERN = "%Y-%m-%d"
-EXPIRY_MILESTONE_RE = re.compile(r'M[0-9]{2,3}\Z')
+EXPIRY_MILESTONE_RE = re.compile(r'M[0-9]{2,4}\Z')
 
 class Error(Exception):
   pass
diff --git a/tools/metrics/histograms/generate_expired_histograms_array.py b/tools/metrics/histograms/generate_expired_histograms_array.py
index bbf283f96f80..ee96a6fbf822 100755
--- a/tools/metrics/histograms/generate_expired_histograms_array.py
+++ b/tools/metrics/histograms/generate_expired_histograms_array.py
@@ -15,8 +15,8 @@ import extract_histograms
 import merge_xml
 
 _DATE_FILE_RE = re.compile(r".*MAJOR_BRANCH_DATE=(.+).*")
-_CURRENT_MILESTONE_RE = re.compile(r"MAJOR=([0-9]{2,3})\n")
-_MILESTONE_EXPIRY_RE = re.compile(r"\AM([0-9]{2,3})")
+_CURRENT_MILESTONE_RE = re.compile(r"MAJOR=([0-9]{2,4})\n")
+_MILESTONE_EXPIRY_RE = re.compile(r"\AM([0-9]{2,4})")
 
 _SCRIPT_NAME = "generate_expired_histograms_array.py"
 _HASH_DATATYPE = "uint64_t"
-- 
2.19.1

